File: week 1/mario.c
#include "cs50.h"
#include <stdio.h>

// Function prototype
void print_bricks(int h);

int main(void)
{
    int height;

    // Prompt the user for a valid height
    do
    {
        height = get_int("Height: ");
    } while (height < 1);
    
    // Print bricks with the specified height
    print_bricks(height);
}

// Function to print bricks in a pyramid shape
void print_bricks(int h)
{
    for (int i = 1; i <= h; i++)
    {
        for (int j = 0; j < h - i; j++)
        {
            printf(" ");
        }

        for (int j = 0; j < i; j++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: week 6/mario.py
def main():
    # Prompt the user for a valid height
    while True:
        try:
            height = int(input("Height: "))
            if 1 <= height <= 8:
                break
        except ValueError:  # Handle non-integer input gracefully
            pass

    # Print bricks with the specified height
    print_blocks(height)

# Function to print bricks in a pyramid shape
def print_blocks(height):
    for i in range(1, height + 1):
        # Print spaces and bricks for each row
        print(" " * (height - i) + "#" * i)

main()
